### PR TITLE
DRMAAV2-28: Replace all malloc to new and handle error conditions

### PR DESCRIPTION
**Pull request by rampranesh**
#### Issue-ID

* **[DRMAAV2-28](https://pbspro.atlassian.net/browse/DRMAAV2-28)**

#### Problem description
* Replace all malloc to new and handle error conditions

#### Cause / Analysis
* NA

#### Solution description
* NA

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **unit test(s) to my commit**. *(or)*
- [x] I have added  **manual test(s) to the Jira ticket and explained why unit testing is not appropriate** for this case.
- [x] All new and existing automated tests have passed. 
- [x] I have attached **test logs to the Jira ticket** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: test/unittesting/src/JobApiTest.cpp [hunk 1]
```diff
@@ -51,7 +51,7 @@ using namespace std;
 CPPUNIT_TEST_SUITE_REGISTRATION(JobApiTest);
 
 void JobApiTest::TestJobApi() {
-        drmaa2_jsession js1 = drmaa2_create_jsession(strdup("SessionApi"), strdup("Contact"));
+        drmaa2_jsession js1 = drmaa2_create_jsession("SessionApi", "Contact");
```
Comment (mike0042) [inline test/unittesting/src/JobApiTest.cpp]: You are using spaces for indentation on this line. The others use tabs.

File: test/unittesting/src/ReservationApiTest.cpp [hunk 2]
```diff
@@ -51,7 +51,7 @@ using namespace std;
 CPPUNIT_TEST_SUITE_REGISTRATION(ReservationApiTest);
 
 void ReservationApiTest::TestReservationApi() {
-        drmaa2_rsession rs1 = drmaa2_create_rsession(strdup("SessionApi"), strdup("Contact"));
+        drmaa2_rsession rs1 = drmaa2_create_rsession("SessionApi", "Contact");
```
Comment (mike0042) [inline test/unittesting/src/ReservationApiTest.cpp]: You are using spaces for indentation on this line. The others use tabs.

**Comment by rampranesh** [comment]
Hi @mike0042 ,
       I have addressed your comments, please have a look at these changes.

Thanks,
Rampranesh M

**Comment by suresh-thelkar** [comment]
Code changes look good to me.

**Pull request merged**